### PR TITLE
Settings service and view

### PR DESCRIPTION
**Pull request by thomas-brunton**

### CODE REVIEW

**Comment by thomas-brunton** [comment]
@Tsz-Fung-Ryan It will fail the travis build for code coverage because the table detail component has not been reworked yet so it doesn't have enough unit tests to get over the 90% threshold.

But if you find any other bugs or issues I will need to fix those.

File: src/app/table/table-detail/table-detail.component.ts [hunk 1]
```diff
@@ -46,7 +45,11 @@ export class TableDetailComponent implements OnInit {
   }
 
   getDetailsData(): void {
-    this.apiService = this.apiSelectorService.getApi(this.apiSetting);
+    if (this.url === undefined) { //  TODO: Figure out why the detail view sometimes makes random data requests with an undefined url which makes errors in the console
+      console.log('Cancelling detail view request since the component randomly sends requests with an undefined url, SHIT\'S FUCKED YO');
```
Comment (Tsz-Fung-Ryan) [inline src/app/table/table-detail/table-detail.component.ts:49]: please remove the console comment

**Review by Tsz-Fung-Ryan — CHANGES_REQUESTED** [review]
Please resolve issues outlined in the code comments

File: src/app/table/table-detail/table-detail.component.ts (hunk 1 above)
Comment (thomas-brunton) [reply src/app/table/table-detail/table-detail.component.ts]: Fixed

**Comment by Tsz-Fung-Ryan** [comment]
Change reviewed and issue outlined resolved

**Pull request merged**